oxide-auth/src/primitives/registrar.rs

```diff
@@ -291,107 +286,69 @@ pub trait PasswordPolicy: Send + Sync {
     fn check(&self, client_id: &str, passphrase: &[u8], stored: &[u8]) -> Result<(), RegistrarError>;
 }
 
-/// Store passwords using `Pbkdf2` to derive the stored value.
-///
-/// Each instantiation generates a 16 byte random salt and prepends this additionally with the
-/// username. This combined string is then used as the salt using the passphrase as the secret to
-/// derive the output. The iteration count defaults to `65536` but can be customized.
-pub struct Pbkdf2 {
-    /// A prebuilt random, or constructing one as needed.
-    random: Option<SystemRandom>,
-    iterations: NonZeroU32,
-}
-
-impl Default for Pbkdf2 {
-    fn default() -> Self {
-        Pbkdf2 {
-            random: Some(SystemRandom::new()),
-            .. *PBKDF2_DEFAULTS
-        }
-    }
+/// Store passwords using `Argon2` to derive the stored value.
+#[derive(Debug)]
+pub struct Argon2<'a> {
+    secret_key: SecretKey<'a>
```

Comment:
I'd rather avoid the lifetime parameter and specify the internal one as `'static`. Note that there exists a method [`to_owned`](https://docs.rs/argonautica/0.2.0/argonautica/input/struct.SecretKey.html#method.to_owned) so this can be done in any case.

Reply:
I did miss to_owned.

Reply:
`'static` as a lifetime does not mean that there is only one. It means that the secret key is _owned_.

Reply:
Are you sure? If so, my bad then, but the [rustbook](https://doc.rust-lang.org/1.9.0/book/lifetimes.html#static) says this:
> The lifetime named ‘static’ is a special lifetime. It signals that something has the lifetime of the entire program. [...] String literals have the type &'static str because the reference is always alive: they are baked into the data segment of the final binary.

Or am I misunderstanding?

Reply:
Yes, you are. In two ways: It's true that a _reference_ with lifetime `'static` refers to data that lives for the entire program. But if something owns the data, e.g. a `Vec<u8>` does, then it is totally under that types control when the data's lifetime ends and it can decide that it lives for the whole program. This _also_ has by definition a `'static` lifetime. This works:

```rust
fn assert_static<T: 'static>() {}
assert_static::<Vec<u8>>();
```

Secondly, this has nothing to do with uniqueness in any case.

Reply:
Ok, it is still not totally clear to me, but I can see it makes sense.
Last commit removed `'a` and replaced it with `'static` on `SecretKey`